controller/controller_server/sync_lock.py

```diff
@@ -0,0 +1,54 @@
+import threading
+from collections import defaultdict
+
+from controller.common.csi_logger import get_stdout_logger
+from controller.controller_server.errors import ObjectAlreadyProcessingError
+
+logger = get_stdout_logger()
+
+
+class SyncLock:
+    def __init__(self):
+        self._lock = threading.Lock()
+        self._ids_in_use = defaultdict(set)
+        self.lock_key = ''
+        self.object_id = ''
+        self.action_name = ''
+
+    def add_to_ids_in_use(self, lock_key, object_id):
+        self._ids_in_use[lock_key].add(object_id)
+
+    def __call__(self, lock_key, object_id, action_name):
+        self.lock_key = lock_key
+        self.object_id = object_id
+        self.action_name = action_name
+        return self
+
+    def __enter__(self):
+        if self.lock_key:
+            self.add_object_lock(self.lock_key, self.object_id, self.action_name)
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if self.lock_key:
+            self.remove_object_lock(self.lock_key, self.object_id, self.action_name)
+
+    def add_object_lock(self, lock_key, object_id, action_name):
+        logger.debug(
+            ("Lock for action: {}, Try to acquire lock for {}: {}".format(action_name, lock_key, object_id)))

```

Comment:
as I [wrote](https://github.com/IBM/ibm-block-csi-driver/pull/443#discussion_r791868466), no need to write "lock" more than once per message.
```suggestion
            ("Trying to acquire lock for action {} with {}: {}".format(action_name, lock_key, object_id)))

```
same for `release lock`, `Succeed to acquire`, etc...

Reply:
done